src/main/scala/junctions/crossing.scala

```diff
@@ -91,3 +90,46 @@ object ClockToSignal {
     c2s.io.signal_out
   }
 }
+
+object LevelCrossing {
+  class SynchronizerBackend[T <: Data](typ: T, sync: Int, _clock: Clock) extends Module(Some(_clock)) {
```

Comment:
Yep, that's exactly the purpose. It's to prevent this component that you are creating from being abused. If someone wants to say "in this case, I understand the implications, and I'm going to build a wrapper around this", they can. Or you can provide a wrapper that does what you're doing here, but give it some appropriately Warning-ful/ descriptive name that indicates what it should be used for.